libxrdp/xrdp_rdp.c

```diff
@@ -36,25 +36,37 @@
 
 /*****************************************************************************/
 static int APP_CC
-xrdp_rdp_read_config(struct xrdp_client_info *client_info)
+xrdp_rdp_read_config(char *xrdp_ini_file, struct xrdp_client_info *client_info)
 {
     int index = 0;
     struct list *items = (struct list *)NULL;
     struct list *values = (struct list *)NULL;
     char *item = (char *)NULL;
     char *value = (char *)NULL;
-    char cfg_file[256];
-
-    /* initialize (zero out) local variables: */
-    g_memset(cfg_file, 0, sizeof(char) * 256);
+    char *p, xrdp_ini_dir[256];
 
     items = list_create();
     items->auto_free = 1;
     values = list_create();
     values->auto_free = 1;
-    g_snprintf(cfg_file, 255, "%s/xrdp.ini", XRDP_CFG_PATH);
-    DEBUG(("cfg_file %s", cfg_file));
-    file_by_name_read_section(cfg_file, "globals", items, values);
+    DEBUG(("xrdp_ini_file %s", xrdp_ini_file));
+
+    g_strncpy(xrdp_ini_dir, xrdp_ini_file, 255);
+    p = (char*)g_strrchr(xrdp_ini_dir, '/');
+    if (p == NULL) {
+        log_message(LOG_LEVEL_ERROR, "cannot calculate dirnane of: %s", xrdp_ini_file);
```

Comment:
here actually should be ```assert``` and validation that ```--config``` contains '/' where command line is parsed